src/sourmash/tax/tax_utils.py

```diff
@@ -831,7 +879,7 @@ class TEXT,
             raise ValueError(f"taxonomy table already exists in '{filename}'")
 
         # follow up and create index
-        cursor.execute("CREATE UNIQUE INDEX taxonomy_ident ON taxonomy(ident);")
+        cursor.execute("CREATE UNIQUE INDEX sourmash_taxonomy_ident ON sourmash_taxonomy(ident);")
```

Comment:
```suggestion
        cursor.execute("CREATE UNIQUE INDEX IF NOT EXISTS sourmash_taxonomy_ident ON sourmash_taxonomy(ident);")
```

Reply:
Running `sourmash lca index ....` twice with the same parameters will try to open and create tables and then threw exceptions. The failed table creation exception was handled in https://github.com/sourmash-bio/sourmash/blob/9e1ea60626a297c82c868e748ab7b0dcc16b8269/src/sourmash/tax/tax_utils.py#L879 which will prevent the process to complete. 
I think it would be better if `lca index` can detect that there's an already saved sqlite database and no need to re-run the command.

If `CREATE TABLE IF EXISTS` was added to that line, running `lca index` will open and update the same sqlite file and that will change the file content from the first run with now any thrown errors.

Reply:
I think for SQL databases, you are 100% right :). But `sourmash lca index` needs to handle JSON databases too, and there are also some issues around the logic for appending that would not be handled well in the JSON case and might require a lot of extra checks.

I ended up going a different route in c443ae10 -
* we now forbid overwriting / appending to existing LCA database;
* I added command-line and API tests for LCA database saving x 2 (all named `twice`);